src/main/java/de/baleipzig/iris/logic/worker/ImageWorker.java

```diff
@@ -0,0 +1,135 @@
+package de.baleipzig.iris.logic.worker;
+
+import de.baleipzig.iris.ImageType;
+import de.baleipzig.iris.logic.converter.ImageConverter;
+import de.baleipzig.iris.model.image.IImage;
+import de.baleipzig.iris.persistence.entity.image.ImageEntity;
+import de.baleipzig.iris.persistence.repository.IImageEntityRepository;
+import lombok.RequiredArgsConstructor;
+import org.apache.commons.io.FilenameUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.*;
+import java.util.stream.Stream;
+
+@Component
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class ImageWorker implements IImageWorker {
+
+    private final Map<ImageType, List<IImage>> allImagesAsMap = new HashMap<>();
+
+    private final IImageEntityRepository repository;
+
+    int count;
+    List<ImageEntity> imageEntities;
+
+    @Override
+    public long countImagesByType(ImageType imageType) {
+        return repository.countByImageType(imageType.toString());
+    }
+
+    @Override
+    public List<IImage> loadRandomImagesByType(int imageCount, ImageType imageType) {
+
+        List<IImage> allImagesByType = loadAllImagesByType(imageType);
+        int imageEntityCount = allImagesByType.size();
+        imageCount = Math.min(imageCount, imageEntityCount);
+
+        Collections.shuffle(allImagesByType);
+
+        return allImagesByType.subList(0, imageCount);
+    }
+
+    @Override
+    public List<IImage> loadAllImagesByType(ImageType imageType) {
+        synchronized (allImagesAsMap) {
```

Comment:
der block kann pro auch bei multithreading nur einmal betreten werden. jeder aufruf auf den integrierten server laeuft in seperatem thread. der worker existiert allerdings nur einmal. ungeschuetzte cache variable kann bei parallelem zugriff zu unschoenen effekten fuehren.